### DISCUSSION TITLE
VCPKG include directories

### DISCUSSION THREAD

**Discussion by jr0me** [microsoft/vcpkg#40307] [General]
I'm working on a project using CMake and Vcpkg, this project includes other subdirectories with their own set of CMake files and dependencies. As for Vcpkg I'm working on manifest mode, and including the vcpkg cmake toolchain before setting the project. For the subdirectories cmake files, I pointed the `VCPKG_MANIFEST_DIR` and `CMAKE_TOOLCHAIN_FILE` relative to their paths, meaning they have to point upwards to the root project path, where they are located.

I've found that doing it this way makes all of the project child subdirectories CMake projects include the vcpkg include directory by default (ie "build/vcpkg_installed/arm64-linux/include"). That has two consequences, I don't need to explicitly call "target_include_directories" and all includes from all dependencies are available to all targets, both things that I don't want to happen.

Interestingly enough from the documentation of Vcpkg I see `vcpkg does not automatically add any include or links paths into your project.`.

Is my setup wrong or is this a limitation of Vcpkg?

**Comment by dg0yt** [comment]
You need to look for the interface of the imported targets you link to.

**Reply by jr0me** [reply]
Hi, thanks for your reply.

Unfortunately, I'm not sure I understand your suggestion, my problem is that I actually don't have to look up for anything.

If the top level project needs dependency A, and child project needs B, they all end up in the path `build/vcpkg_installed/arm64-linux/include`. When I inspect the `LINK_LIBRARIES` property of each target (parent and children) they all include that path. So, top level sees both A and B, child projects too.

Header only libraries will compile, while non header only will still need to be linked against.

My expectation would be that the path `vcpkg_installed/arm64-linux/include` isn't set. And only `vcpkg_installed/arm64-linux/include/A` or `vcpkg_installed/arm64-linux/include/B` are set using `target_link_libraries` or `target_include_directories` for specific targets and their corresponding dependencies.

Is there a way to accomplish that?

**Reply by dg0yt** [reply]
Cf. https://github.com/microsoft/vcpkg/discussions/26518#discussioncomment-10252566. As longs as msbuild users don't get `include/A` and `include/B` added automatically, I wouldn't expect to see separate install prefixes for separate packages. So there is no way to accomplish this.
